### PR TITLE
345 remove batches from bigmap imp2

### PR DESCRIPTION
**Pull request by glottologist**
Alternative implementation using batches to hold their own count of holdings

### CODE REVIEW

File: batcher/batcher.mligo [hunk 1]
```diff
@@ -40,6 +40,11 @@
 [@inline] let cannot_update_scale_factor_to_more_than_the_maximum : nat          = 136n
 [@inline] let cannot_remove_swap_pair_that_is_not_disabled : nat                 = 137n
 [@inline] let token_name_not_in_list_of_valid_tokens : nat                       = 138n
+[@inline] let no_orders_for_user_address : nat                                   = 139n
+[@inline] let cannot_cancel_orders_for_a_batch_that_isn_not_open : nat           = 140n
```
Comment (aguillon) [inline batcher/batcher.mligo:44]: typo btw

**Review by aguillon — APPROVED** [review]
This one makes more sense to me, except for the fee refunding, but it's a detail. What do you think?

**Comment by glottologist** [comment]
> This one makes more sense to me, except for the fee refunding, but it's a detail. What do you think?

I agree, this imp is better.

File: batcher/batcher.mligo (hunk 1 above)
Comment (glottologist) [reply batcher/batcher.mligo:44]: This is from the cancellations branch. I have fixed there

**Pull request merged**